src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -51,15 +59,68 @@ const Zmenu = (props: ZmenuProps) => {
 
   const destroyZmenu = () => {
     dispatch(changeHighlightedTrackId(''));
-    setZmenus && setZmenus(pickBy(zmenus, (value, key) => key !== props.id));
+    setZmenus &&
+      setZmenus(pickBy(zmenus, (value, key) => key !== props.zmenuId));
   };
 
   const direction = chooseDirection(props);
   const toolboxPosition =
     direction === Direction.LEFT ? ToolboxPosition.LEFT : ToolboxPosition.RIGHT;
 
+  const { content, variety } = props.payload;
+
+  const transcripts = content.filter(
+    (f) => f.metadata.type === ZmenuFeatureType.TRANSCRIPT
+  ) as ZmenuContentTranscript[];
+  const genes = content.filter(
+    (f) => f.metadata.type === ZmenuFeatureType.GENE
+  ) as ZmenuContentGene[];
```

Comment:
You are doing the same on line 98 using the more appropriate `find` method (a gene can be found as a feature of type gene whole id equals the gene id of the transcript)

Reply:
My understanding is that we will still have multiple genes and transcripts until gene bumping is out.

Reply:
But we can just grab the first one.

Reply:
That's done